src/core/meshcop/leader.hpp

```diff
@@ -26,11 +26,131 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if OPENTHREAD_MTD
-#include "leader_mtd.hpp"
-#elif OPENTHREAD_FTD
-#include "leader_ftd.hpp"
-#else
-#error "Please define OPENTHREAD_MTD=1 or OPENTHREAD_FTD=1"
-#endif
+/**
+ * @file
+ *   This file includes definitions for a MeshCoP Leader.
+ */
+
+#ifndef MESHCOP_LEADER_HPP_
+#define MESHCOP_LEADER_HPP_
+
+#include <coap/coap_client.hpp>
+#include <coap/coap_server.hpp>
+#include <common/timer.hpp>
+#include <meshcop/tlvs.hpp>
+#include <net/udp6.hpp>
+#include <thread/mle.hpp>
+
+namespace Thread {
+namespace MeshCoP {
+
+OT_TOOL_PACKED_BEGIN
+class CommissioningData
+{
+public:
+    uint8_t GetLength(void) const {
```

Comment:
FWIW, the current [STYLE_GUIDE.md](https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md) contains the following statement:

> - Inline functions should be used judiciously.
>   - The use of code in headers and, more specifically, the use of the non-local scope inline functions should be avoided. Exception: Simple setters and getters are fine since the compiler can efficiently optimize these and make their overhead as low as a direct data member access.

I agree this probably assumes that developers are always striving to do the right thing and performing the kind of evaluation @chshu has done above.